addons/product/models/decimal_precision.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, _
+from odoo.exceptions import ValidationError
+from odoo.tools.float_utils import float_compare
+
+
+class DecimalPrecision(models.Model):
+    _inherit = 'decimal.precision'
+
+    @api.constrains('digits')
+    def _check_main_currency_rounding(self):
+        decimal_precision = self.search([('name', 'like', 'Account')], limit=1)
+        main_currency = self.env.user.company_id.currency_id
+        if decimal_precision and self == decimal_precision and \
```

Comment:
Use self.ensure_one() to verify that the self contains only one record or make the method to work on multiple records.